web/event_utils.js

```diff
@@ -172,33 +172,56 @@ class EventBus {
 /**
  * NOTE: Only used to support various PDF viewer tests in `mozilla-central`.
  */
-class AutomationEventBus extends EventBus {
+class FirefoxEventBus extends EventBus {
+  #externalServices;
+
+  #globalEventNames;
+
+  #isInAutomation;
+
+  constructor(globalEventNames, externalServices, isInAutomation) {
+    super();
+    this.#globalEventNames = globalEventNames;
+    this.#externalServices = externalServices;
+    this.#isInAutomation = isInAutomation;
+  }
+
   dispatch(eventName, data) {
     if (typeof PDFJSDev !== "undefined" && !PDFJSDev.test("MOZCENTRAL")) {
-      throw new Error("Not implemented: AutomationEventBus.dispatch");
+      throw new Error("Not implemented: FirefoxEventBus.dispatch");
     }
     super.dispatch(eventName, data);
 
-    const detail = Object.create(null);
-    if (data) {
-      for (const key in data) {
-        const value = data[key];
-        if (key === "source") {
-          if (value === window || value === document) {
-            return; // No need to re-dispatch (already) global events.
+    if (this.#isInAutomation) {
```

Comment:
Given that this special-case is only needed for mozilla-central tests, would it be (at all) feasible to re-factor those event listeners to instead use the new `globalEventNames` functionality?

Reply:
Good idea but let's do that in a follow-up because it'll induce some unrelated (w.r.t. the initial goal of this patch: help omc team) changes in the m-c tests.